### DISCUSSION TITLE
Is there a way to disable the auto-generation of config/ for external crates?

### DISCUSSION THREAD

**Discussion by pmunts** [alire-project/alire#1319] [Q&A]
If an external crate happens to have been cross-compiled, the contents of the auto-generated `config/` are very misleading, and I would like to suppress it.

I have tried

```
[configuration]
disabled = true
```

in various places in the external crate `.toml` file, but it doesn't seem to be allowed.

**Comment by mosteo** [comment]
I'm not following how a crate can be external and have sources and a config folder. In any case, for regular crates, the snippet you showed will disable generation of `config/` contents, but it won't delete anything previously existing; you have to delete it by hand.

**Reply by pmunts** [reply]
I noticed that when a project includes an external crate, that crate's subdirectory under `alire/cache/dependencies/` contains an empty subdirectory `alire/` (that should probably contain `alire.toml.upstream` like a regular crate, a separate issue) and a subdirectory `config/` that could possibly contain misleading information.  Since regular crates can suppress `config/`, I wondered if the same would be or ought to be possible for external crates.

**Reply by mosteo** [reply]
I see, and indeed I can reproduce it.

Although arguably there is some possibly useful info in the generated files, like the version in use, it's true that there are build switches and stuff that doesn't make sense in that case. I'm not sure right now if it's better to keep it as-is for regularity or remove it.
